File: web-client/chinese-chess/src/scene/play/rule/ChessPos.ts
/**
 * 棋子位置
 */
export default class ChessPos {
    /**
     * 棋盘纵坐标，从0开始数
     */
    readonly row: number;

    /**
     * 棋盘横坐标，从0开始数
     */
    readonly col: number;

    constructor(row: number, col: number) {
        this.row = row;
        this.col = col;
    }

    copy() {
        return new ChessPos(this.row, this.col);
    }
}

File: server/src/main/java/io/github/hulang1024/chinesechessserver/play/rule/ChessPos.java
package io.github.hulang1024.chinesechessserver.play.rule;

import lombok.Data;

@Data
public class ChessPos {
    public int row;
    public int col;

    public ChessPos(int row, int col) {
        this.row = row;
        this.col = col;
    }
}
